build.gradle

```diff
@@ -78,76 +78,99 @@ project(':configurator') {
-            try{
-                out << downloadUrl.openStream()
-                out.close()
-            }
-            catch(IOException e){
-                e.printStackTrace()
-            }
-            finally {
-                file.close()
-            }
+        if(!jdkDownloadDir.exists()) {
+            jdkDownloadDir.mkdirs()
         }
-    }
-    // Check the hash of JRE
-    doLast {
-        def digest = MessageDigest.getInstance("SHA-256");
-        def buffer = new byte[1024]
-        def len
-        def inputStream = new FileInputStream(jreDir)
-        try {
-            while ((len = inputStream.read(buffer, 0, 1024)) != -1) {
-                digest.update(buffer, 0, len)
+        if(!jdkFile.exists()){
+            new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/${jdkFileName}")
+                    .withInputStream {
+                jdkFile.newOutputStream() << it
+            }
+            def digest = MessageDigest.getInstance("SHA-256");
+            jdkFile.eachByte(1024) {buff,len ->
+                digest.update(buff, 0, len)
             }
-            BigInteger bigInt = new BigInteger(1, digest.digest())
-            String hashValue = bigInt.toString(16)
-            if(!hashValue.equals(originalFileHash)){
-                throw new Exception("Please check if JRE is downloaded securely")
+            final hash = digest.digest().encodeHex() as String
+            if(hash != originalFileHash){
+                throw new RuntimeException("Hash mismatch for downloaded JRE. File may have been modified.")
             }
-        } catch (IOException e) {
-            e.printStackTrace()
-        } finally {
-            inputStream.close()
         }
     }
 }
 
-task createPackages(dependsOn:downloadJre) {
-    //Sub tasks are dynamically created by iterating systemNames
+task unzipJdk(dependsOn:downloadJdk ,type: Copy) {
+    description "Unzip JDK only when it is firstly downloaded"
+    from tarTree(resources.gzip(jdkFile))
+    into jdkDownloadDir
 }
-systemNames.each { sys ->
-    File sysPackageDir = new File("$projectDir/packages")
-    File sysScriptDir = new File("$projectDir/launcher-scripts/$sys-launcher")
-    task "packageFor$sys"(type: Zip){
-        doFirst {
-            if(!sysPackageDir.exists()){
-                sysPackageDir.mkdir()
-            }
-        }
-        archiveFileName = "admin-console-package-for-"+sys+".zip"
-        destinationDirectory = sysPackageDir
+unzipJdk.onlyIf {
+    !jdkExtractDir.exists()
+}
+
+task createPackages(dependsOn:unzipJdk) {
+    description "Create three packages for Linux, Windows and Mac"
+    //This task also depends on other three tasks which are created by iterating osNames
+}
+
+osNames.each { sys ->
+    def packageDir = new File("$buildDir/distributions/packages")
+    def(taskType, fileExtension, fileCompression, scriptFile) = setPackageProperties(sys.toString())
```

Comment:
Please add a space between `def` and `(`.

Rather than `sys.toString()` do `sys as String`, but better yet, up on line 129 just do `osNames.each { String sys ->`.